unit_test/sparse/Test_Sparse_block_gauss_seidel.hpp

```diff
@@ -404,18 +374,30 @@ void test_block_gauss_seidel_rank2(lno_t numRows, size_type nnz,
   // device::execution_space::finalize();
 }
 
-#define EXECUTE_TEST(SCALAR, ORDINAL, OFFSET, DEVICE)                                     \
-  TEST_F(                                                                                 \
-      TestCategory,                                                                       \
-      sparse##_##block_gauss_seidel_rank1##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) { \
-    test_block_gauss_seidel_rank1<SCALAR, ORDINAL, OFFSET, DEVICE>(                       \
-        500, 500 * 10, 70, 3);                                                            \
-  }                                                                                       \
-  TEST_F(                                                                                 \
-      TestCategory,                                                                       \
-      sparse##_##block_gauss_seidel_rank2##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) { \
-    test_block_gauss_seidel_rank2<SCALAR, ORDINAL, OFFSET, DEVICE>(                       \
-        500, 500 * 10, 70, 3);                                                            \
+#define EXECUTE_TEST(SCALAR, ORDINAL, OFFSET, DEVICE)                                    \
```

Comment:
Would it make sense to add tests where the size of the matrix is 0 rows or where some rows are empty and such to assess robustness of the overall Gauss-Sedeil?
@brian-kelley does that make sense?

Reply:
Yes that's a good idea, and the non-block GS has a test like that (test_sgs_zero_rows) since there have been bugs in the past.

Reply:
Thank you for pointing that out. I added `test_sgs_zero_rows` to block unit test and - on [the other commit](https://github.com/kokkos/kokkos-kernels/pull/1232/commits/257e04797f7e5b9ad479b8f62af1c31b2deb5c62) - extended it with 0-nnz matrices (all-zeros matrix with empty rows). Let me know if you don't like it or have other suggestions.

Reply:
(I've seen empty rows triggering different types of bugs than 0-size matrices)